File: hw/bonilla/u1/hw14SaveinJsonChickenFarmSimulator/ChickenFarmSimulatorPython/ChickenCoop.py
class ChickenCoop:
    id_counter = 1  

    def __init__(self, name):
        self.id = ChickenCoop.id_counter
        ChickenCoop.id_counter += 1
        self.name = name
        self.chickens = []

    def add_chicken(self, chicken):
        self.chickens.append(chicken)

    def get_chickens(self):
        return self.chickens

    def get_next_chicken_id(self):
        return len(self.chickens) + 1

    def __str__(self):
        return f"ChickenCoop {{ id={self.id}, name={self.name}, numberOfChickens={len(self.chickens)} }}"

File: hw/bonilla/u1/hw14SaveinJsonChickenFarmSimulator/ChickenFarmSimulatorJava/ChickenCoop.java
package ChickenFarmSimulatorJava;

import java.util.ArrayList;
import java.util.List;

public class ChickenCoop {
    private static int idCounter = 1;

    private int id;
    private String name;
    private List<Chicken> chickens;

    public ChickenCoop(String name) {
        this.id = idCounter++;
        this.name = name;
        this.chickens = new ArrayList<>();
    }

    public void addChicken(Chicken chicken) {
        chickens.add(chicken);
    }

    public List<Chicken> getChickens() {
        return chickens;
    }

    public int getNextChickenId() {
        return chickens.size() + 1;
    }

    @Override
    public String toString() {
        return "ChickenCoop { " +
                "id=" + id +
                ", name='" + name + '\'' +
                ", numberOfChickens=" + chickens.size() +
                " }";
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}